analyzers/src/SonarAnalyzer.CSharp/Rules/MemberShadowsOuterStaticMember.cs

```diff
@@ -43,12 +35,17 @@ public sealed class MemberShadowsOuterStaticMember : SonarDiagnosticAnalyzer
                 {
                     var innerClassSymbol = (INamedTypeSymbol)c.Symbol;
                     var containerClassSymbol = innerClassSymbol.ContainingType;
-                    if (!innerClassSymbol.IsClassOrStruct() || !containerClassSymbol.IsClassOrStruct())
+
+                    if (!IsValidType(innerClassSymbol) || !IsValidType(containerClassSymbol))
```

Comment:
I suggest merging the two Ifs that just return

Reply:
i think this condition:
```csharp
if (innerClassSymbol
      .GetMembers()
      .Where(x => !x.IsImplicitlyDeclared && !IsStaticAndVirtualOrAbstract(x))
      .ToList() 
    is not { Count: not 0 } members)
```
Is slightly too complicated, since it has both the double negation `not` and a `Count: not 0`.

I think it is better to leave it as-is, if you don't mind it.